fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biomes/InternalBiomeData.java

```diff
@@ -0,0 +1,184 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.biomes;
+
+import com.google.common.base.Preconditions;
+
+import it.unimi.dsi.fastutil.objects.Object2DoubleMap;
+import it.unimi.dsi.fastutil.objects.Object2DoubleOpenHashMap;
+import net.fabricmc.fabric.api.biomes.v1.OverworldClimate;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.Biomes;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Lists and maps for internal use only! Stores data that is used by the various mixins into the world generation
+ */
+public final class InternalBiomeData {
+
+	private InternalBiomeData() {
+	}
+
+	private static final Map<Biome, WeightedBiomePicker> OVERWORLD_HILLS_MAP = new HashMap<>();
+	private static final Map<Biome, WeightedBiomePicker> OVERWORLD_SHORE_MAP = new HashMap<>();
+	private static final Map<Biome, WeightedBiomePicker> OVERWORLD_EDGE_MAP = new HashMap<>();
+	private static final Map<Biome, Biome> OVERWORLD_RIVER_MAP = new HashMap<>();
+	private static final Map<Biome, VariantTransformer> OVERWORLD_VARIANT_TRANSFORMERS = new HashMap<>();
+	
+	private static final Map<OverworldClimate, Object2DoubleMap<Biome>> OVERWORLD_BIOME_WEIGHT_TOTALS = new HashMap<>();
+	private static final Map<OverworldClimate, List<BiomeEntry>> OVERWORLD_BASE_BIOMES = new HashMap<>();
+	protected static final EnumMap<OverworldClimate, Double> OVERWORLD_MODDED_WEIGHT_TOTALS = new EnumMap<>(OverworldClimate.class);
+
+	private static final List<Biome> OVERWORLD_INJECTED_BIOMES = new ArrayList<>();
+
+	private static final Set<Biome> SPAWN_BIOMES = new HashSet<>();
+	
+	public static void addOverworldBaseBiome(OverworldClimate climate, Biome biome, double weight) {
+		Preconditions.checkArgument(climate != null && biome != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
+		Preconditions.checkArgument(weight > 0, "Weight is zero or negative (must be positive)");
+		OVERWORLD_BASE_BIOMES.computeIfAbsent(climate, k -> new ArrayList<>()).add(new BiomeEntry(biome, weight, climate));
+		OVERWORLD_INJECTED_BIOMES.add(biome);
+		
+		Object2DoubleMap<Biome> climateMap = OVERWORLD_BIOME_WEIGHT_TOTALS.get(climate);
+		double currentWeight = climateMap.computeIfAbsent(biome, (b) -> Double.valueOf(0));
+		climateMap.replace(biome, currentWeight + weight);
+	}
+
+	public static void addOverworldHillsBiome(Biome parent, Biome hills, int weight) {
+		Preconditions.checkArgument(parent != null && hills != null, "One or both arguments are null");
+		Preconditions.checkArgument(!Double.isNaN(weight), "Weight is NaN");
```

Comment:
There seems to be a bit of a confusion here. `int` values cannot be NaN, but is the weight supposed to be an int ? An explanation as to why the hills/shore/edge maps and the base biomes map work differently would be helpful.